src/routes/dashboard/helpers/channels.ts

```diff
@@ -0,0 +1,75 @@
+import type { Question, Questions } from '../types'
+
+export function sortChannels(questions: Question[]) {
+  const counts = questions.reduce((count, question) => {
+    const key = question.channelName
+    return count[key] ? ++count[key] : (count[key] = 1), count
+  }, {})
+
+  return Object.entries(counts)
+    .map(([channelName, count]) => {
+      return { group: channelName, count }
+    })
+    .sort((a, b) => a.group.localeCompare(b.group))
+}
+
+const percentColors = [
+  { pct: 0, color: { r: 0xff, g: 0x00, b: 0 } },
+  { pct: 50, color: { r: 0xff, g: 0xff, b: 0 } },
+  { pct: 100, color: { r: 0x00, g: 0xff, b: 0 } },
+]
+
+const getColor = (pct: number) => {
+  for (var i = 1; i < percentColors.length - 1; i++) {
+    if (pct < percentColors[i].pct) {
+      break
+    }
+  }
+  const lower = percentColors[i - 1]
+  const upper = percentColors[i]
+  const range = upper.pct - lower.pct
+  const rangePct = (pct - lower.pct) / range
```

Comment:
```suggestion
  const rangePct = Math.floor((pct - lower.pct) / range)
```

The floating point issue may not apply here, but just in case https://floating-point-gui.de/

https://www.delftstack.com/howto/javascript/integer-division-javascript/#:~:text=In%20JavaScript%2C%20we%20can%20divide,floor()%20or%20Math.